### PR TITLE
BLD: Make cblas detection work on windows with mingw32.

### PR DESCRIPTION
**Pull request by charris**
Currently the compile only checks with msvc, which leads to problems as it
is not installed in the usual mingw setup.

### CODE REVIEW

**Comment by charris** [comment]
@juliantaylor Does this look right? Will it work for the various SSE combinations?

File: numpy/distutils/system_info.py [hunk 1]
```diff
@@ -1702,10 +1704,29 @@ def has_cblas(self):
                 c.compile([src], output_dir=tmpdir,
                           include_dirs=self.get_include_dirs())
                 res = True
-            except distutils.ccompiler.CompileError:
+            except DistutilsError:
```
Comment (charris) [inline numpy/distutils/system_info.py]: Could use the more specific `DistutilsPlatformError` to try the fallback.

**Comment by charris** [comment]
It is strange that at the point `has_cblas` is called, the actual compiler to use is unknown.

**Comment by charris** [comment]
Perhaps the easiest thing to do is remove this check. It was intended as a fall back check if all else failed and removing it would probably cause no problems.

**Comment by charris** [comment]
On second thought, removing the check looks like the better fix, so closingin this. CBLAS detection problems are probably better handled in `site.cfg`.

**Comment by carlkl** [comment]
My experience is, that during build with mingw32 or mingwpy (mingw-w64) the function ```numpy.distutils.ccompiler.new_compiler()``` may return ```'msvc'``` instead of ```'mingw32'```.
Thus the compiler detection in case of an existing MS compiler on the build machine can be wrong.

The reason is, that the function ```mingw32()``` in ```distutils.misc_util``` is insufficient and may return a wrong result. This is due to the fact hat neither ```OSTYPE``` or ```MSYSTEM``` may be set and thus returns ```FALSE``` even in case of a mingw32 build.

IMHO ```mingw32()``` should only rely on the setting of ```%USERPROFILE\%pydistutils.cfg``` or ```distutils.cfg``` and/or the commandline switch ```--compiler=mingw32``` during numpy build. 

Is there an obvious way to detect this settings correctly?

**Comment by charris** [comment]
@carlkl I agree that mingw() needs some work. For numpy-vendor I ended up defining `OSTYPE='msys'` but better detection is certainly desirable. A quick google didn't turn up any easy way to get the result of ```--compiler=mingw32``` apart from parsing the command line. If the `cmd` variable is available the compiler name can be used, but that requires tracking down where that variable is set.

**Comment by carlkl** [comment]
@charris, unfortunatly ```distutils.sysconfig.get_config_var("CC")``` returns ```None``` in case of mingw builds. I can parse pydistutils.cfg (if available) with:
```python
from distutils.dist import Distribution
dist = Distribution()
dist.parse_config_files()
dist.get_option_dict('build')
>>> {'compiler': ('PATH_TO\\pydistutils.cfg', 'mingw32')}
```
maybe ```dist.parse_command_line()``` works for the setup.py commandline; I have to check.

**Pull request closed without merging**